* bug fix in photon plan import --> use MU from dcm

markbangert opened the pull request.

AlexS1992 commented:
@markbangert Your changes in photonDoseCalc look okay to me, except the fact that now we cant use bixel based dose calculation and a primary fluence together and we cant use field based dose calc without primary fluence. Essentially you could remove useCustomPrimFluenceBool because it has now the same meaning as bixelWidth = "field". Is that what you intended?

Concerning the changes in the DICOM import:
As a first change to start things of this looks okay, but I am not yet sure how the correct field weight factor would look like. If you want to provide a fix for the problem during dose calculation you could merge this now and we can correct the function later on.

markbangert commented:
thanks for your feedback @AlexS1992 ! I will keep looking into the dicom import. 

regarding your comment on the primary fluence: I think it makes sense to allow field based calculation only in combination with a custom primary fluence. For a bixel-based dose calculation, however, there should be the possibility to switch between the two options and I think that can still be done by manually setting useCustomPrimFluenceBool in line 107 to true. That is exactly the way i was implemented earlier. There you also had to make this modification by hand. I just updated the function including a comment to make this clear. Or am I overseeing something here?

markbangert commented on matRad_calcPhotonDose.m:
```diff
@@ -220,29 +226,27 @@
         bixelsPerBeam = bixelsPerBeam + 1;
     
         % convolution here if custom primary fluence OR field based dose calc
-        if useCustomPrimFluenceBool || strcmp(num2str(pln.bixelWidth),'field')
```
Good call! Will do so.

markbangert commented on matRad_calcPhotonDose.m:
```diff
@@ -99,7 +99,13 @@
 
 % toggle custom primary fluence on/off. if 0 we assume a homogeneous
 % primary fluence, if 1 we use measured radially symmetric data
-useCustomPrimFluenceBool = 0;
```
Good call! Will do so.

markbangert commented:
@AlexS1992 I just updated this pull request. 

1. I think I found some discrepancy during the import of the isocenter which might explain the observed shifts.
2. Also I was able to correct a couple of bugs in the code for the field import - mostly I incorporated them myself earlier. However, there is still something wrong somewhere in the code yielding wrong dosimetry with recalculated plans. The dose has a global offset of more than 10% at the moment...

markbangert commented:
@becker89 I know you are busy with more important stuff at the moment but I was wondering if you are following this issue. Considering that everything (isocenter/beam geometry) worked out just fine leaves me wonder if we are rather implementing new bugs than getting rid of old ones.

Do you still have the pipeline up and running you used for the particle dose calculation validation? Could you rerun a DICOM particle plan import and recalculate the dose real quick to double check if everything is correct with the proposed changes? Please give honest feedback if that would take too much time for you right now then I will try to reproduce the particle results myself!

becker89 commented:
Oh no its fine - I will have a look at it tomorrow morning

markbangert commented:
I did some tests yesterday that basically make a test for particles
obsolete. I compared the isocenter computations from the two different
branches and now everything is consistent. I think you don't need to run
the tests.

Am 16.02.2017 6:21 vorm. schrieb "becker89" <notifications@github.com>:

> Oh no its fine - I will have a look at it tomorrow morning
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/e0404/matRad/pull/131#issuecomment-280234285>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJr8fa8OKSDPaGcUxjVnw5w5YbWBaympks5rc9zugaJpZM4L5cVn>
> .
>

becker89 commented:
Ok great - then I will not run additional tests.
